extension/chrome/settings/modules/security.ts

```diff
@@ -7,122 +7,135 @@ import { Store } from '../../../js/common/platform/store.js';
+    $('.password_message_language').val(storage.outgoing_language || 'EN');
+    await this.renderPPOptionsIfStoredInSession();
+    await this.renderSubscriptionDetails();
+  }
 
-  const [primaryKi] = await Store.keysGet(acctEmail, ['primary']);
-  Assert.abortAndRenderErrorIfKeyinfoEmpty(primaryKi, false);
-  if (!primaryKi) {
-    return; // added do_throw=false above + manually exiting here because security.htm can indeed be commonly rendered on setup page before setting acct up
+  setHandlers() {
+    $('.action_change_passphrase').click(this.setHandler(() => Settings.redirectSubPage(this.acctEmail, this.parentTabId, '/chrome/settings/modules/change_passphrase.htm')));
+    $('.action_test_passphrase').click(this.setHandler(() => Settings.redirectSubPage(this.acctEmail, this.parentTabId, '/chrome/settings/modules/test_passphrase.htm')));
+    $('#hide_message_password').change(this.setHandler((el) => this.hideMsgPasswordHandler(el)));
+    $('.password_message_language').change(this.setHandler(() => this.onMsgLanguageUserChange()));
   }
 
-  const storage = await Store.getAcct(acctEmail, ['hide_message_password', 'outgoing_language']);
-  const authInfo = await Store.authInfo(acctEmail);
+  private async renderPPOptionsIfStoredInSession() {
+    let storedInSession = false;
+    const keys = await Store.keysGet(this.acctEmail);
+    for (const key of keys) {
+      if (await Store.passphraseGet(this.acctEmail, key.longid, true)) {
+        storedInSession = true;
+        break;
+      }
+    }
+    if (storedInSession) {
+      $('.forget_passphrase').css('display', '');
+      $('.action_forget_pp').click(this.setHandler(() => {
+        $('.forget_passphrase').css('display', 'none');
+        $('.passphrase_entry_container').css('display', '');
+      }));
+      $('.confirm_passphrase_requirement_change').click(this.setHandler(async () => {
+        const primaryKiPP = await Store.passphraseGet(this.acctEmail, this.primaryKi!.longid);
+        if ($('input#passphrase_entry').val() === primaryKiPP) {
+          for (const key of keys) {
+            await Store.passphraseSave('local', this.acctEmail, key.longid, undefined);
+            await Store.passphraseSave('session', this.acctEmail, key.longid, undefined);
+          }
+          window.location.reload();
+        } else {
+          await Ui.modal.warning('Pass phrase did not match, please try again.');
+          $('input#passphrase_entry').val('').focus();
+        }
+      }));
+      $('.cancel_passphrase_requirement_change').click(() => window.location.reload());
+      $('#passphrase_entry').keydown(this.setHandler((el, ev) => {
+        if (ev.which === 13) {
+          $('.confirm_passphrase_requirement_change').click();
+        }
+      }));
+    }
+  }
 
-  if (embedded) {
-    $('.change_passhrase_container, .title_container').css('display', 'none');
-    $('.line').css('padding', '7px 0');
+  private async renderSubscriptionDetails() {
```

Comment:
this could be renamed `loadAndRenderPwdEncryptedMsgSettings`

Reply:
Yes, this is better,
Done